pkg/cluster/managed.go

```diff
@@ -47,13 +52,27 @@ func (c *Cluster) testClusterDB(ctx context.Context) (<-chan struct{}, error) {
 // start starts the database, unless a cluster reset has been requested, in which case
 // it does that instead.
 func (c *Cluster) start(ctx context.Context) error {
+	resetFile := filepath.Join(c.config.DataDir, "db", "reset-flag")
 	if c.managedDB == nil {
 		return nil
 	}
 
 	if c.config.ClusterReset {
+		if _, err := os.Stat(resetFile); err != nil {
+			if !os.IsNotExist(err) {
+				return err
+			}
+		} else {
+			return fmt.Errorf("cluster-reset was successfully performed, "+
+				"please remove the cluster-reset flag and start %s normally, "+
+				"if you need to perform another cluster reset, "+
+				"you must first manually delete the %s file",
+				version.Program, resetFile)
+		}
 		return c.managedDB.Reset(ctx, c.clientAccessInfo)
 	}
+	// removing the reset file and ignore error if the file doesnt exist
+	os.Remove(resetFile)
```

Comment:
Could we ever care about the returned error here?

Reply:
no not really, if file doesnt exist which happens on normal start of rke2/k3s it will just ignore it and start etcd